app/scripts/modules/openstack/serverGroup/configure/serverGroupConfiguration.service.js

```diff
@@ -0,0 +1,222 @@
+        var removed = _.xor(matched, current);
+        command.loadBalancers = matched;
+        if (removed.length) {
+          result.dirty.loadBalancers = removed;
+        }
+      }
+      command.backingData.filtered.loadBalancers = newLoadBalancers;
+      return result;
+    }
+
+    function refreshLoadBalancers(command, skipCommandReconfiguration) {
+      return $q.when()
+        .then(cacheInitializer.refreshCache('loadBalancers'))
+        .then(loadBalancerReader.listLoadBalancers('openstack'))
+        .then(function (loadBalancers) {
+          command.backingData.loadBalancers = loadBalancers;
+          if (!skipCommandReconfiguration) {
+            configureLoadBalancerOptions(command);
+          }
+        });
+    }
+
+    function configureLoadBalancers(command) {
+      var result = { dirty: {} };
+      var temp = command.backingData.loadBalancers;
+      var filterlist = _.filter(temp, function(lb) { return (lb.account === command.credentials && lb.region === command.region);});
+
+      command.loadBalancers = getLoadBalancerNames(filterlist);
+      command.viewState.loadBalancersConfigured = true;
+
+      return result;
+    }
+
+    function attachEventHandlers(command) {
+
+      command.regionChanged = function regionChanged() {
+        var result = { dirty: {} };
+        if (command.region && command.credentials) {
+          angular.extend(result.dirty, configureLoadBalancers(command).dirty);
+          angular.extend(result.dirty, configureSecurityGroupOptions(command).dirty);
+        }
+
+        return result;
+      };
+
+      command.credentialsChanged = function credentialsChanged() {
+        var result = { dirty: {} };
+        var backingData = command.backingData;
+        if (command.credentials) {
+          var regionsForAccount = backingData.credentialsKeyedByAccount[command.credentials] || {regions: [], defaultKeyPair: null};
+          backingData.filtered.regions = regionsForAccount.regions;
+          if (!_(backingData.filtered.regions).some({name: command.region})) {
+            command.region = null;
+            result.dirty.region = true;
+          } else {
+            angular.extend(result.dirty, command.regionChanged().dirty);
+          }
+          if (command.region) {
+            angular.extend(result.dirty, configureLoadBalancers(command).dirty);
+          }
+        } else {
+          command.region = null;
+        }
+        return result;
+      };
+    }
+
+    return {
+      configureUpdateCommand: configureUpdateCommand,
+      configureCommand: configureCommand,
+      configureImages: configureImages,
+      configureSecurityGroupOptions: configureSecurityGroupOptions,
+      configureLoadBalancerOptions: configureLoadBalancerOptions,
+      refreshLoadBalancers: refreshLoadBalancers,
+      refreshSecurityGroups: refreshSecurityGroups,
+      getRegionalSecurityGroups: getRegionalSecurityGroups,
```

Comment:
Better to remove all the security group code (these two methods + `configureSecurityGroupOptions`) from this service - should knock the file down by 50 lines or so.